### PR TITLE
Feature/65 as a doctor i want to moderate black list

### PR DESCRIPTION
**Pull request by SergiyKozachok**

### CODE REVIEW

File: InternetHospital.BusinessLogic/Models/FilteredMyPatientsModel.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace InternetHospital.BusinessLogic.Models
+{
+    public class FilteredMyPatientsModel
```
Comment (voznovych) [inline InternetHospital.BusinessLogic/Models/FilteredMyPatientsModel.cs]: There are generic type for returning pagination, even 2 types: FilteredModel and PageModel.

File: InternetHospital.BusinessLogic/Services/DoctorService.cs [hunk 2]
```diff
@@ -248,5 +253,130 @@ private bool FillIllness(IllnessHistoryModel illnessModel, Appointment appointme
...
+        /// Get all patients who had appointments for current doctor
+        /// </summary>
+        /// <param name="doctorId"></param>
+        /// <returns>
+        /// returns a collection of doctor's patients
+        /// </returns>
+        private IQueryable<User> GetDateOfMyPatients(int doctorId)
+        {
+            var patients = _context.Users.Where(x => x.Appointments.Any(a => a.DoctorId == doctorId))
+                .Select(p => new User
```
Comment (voznovych) [inline InternetHospital.BusinessLogic/Services/DoctorService.cs:328]: You can use Automapper here.

File: InternetHospital.BusinessLogic/Services/DoctorService.cs [hunk 3]
```diff
@@ -248,5 +253,130 @@ private bool FillIllness(IllnessHistoryModel illnessModel, Appointment appointme
...
+        /// <summary>
+        /// Convert patients collection from User model to MyPatientModel
+        /// </summary>
+        /// <param name="patients"></param>
+        /// <returns>
+        /// return collection of patients
+        /// </returns>
+        private IQueryable<MyPatientModel> ConvertToViewModel(IQueryable<User> patients)
+        {
+            var _patients = patients.Select(p => new MyPatientModel
```
Comment (voznovych) [inline InternetHospital.BusinessLogic/Services/DoctorService.cs]: You can use automapper and remove this method.

File: InternetHospital.BusinessLogic/Services/DoctorService.cs (hunk 2 above)
Comment (SergiyKozachok) [reply InternetHospital.BusinessLogic/Services/DoctorService.cs:328]: Yes, but I do not want to do this, he uses additional resources, but from him the code is much less readable. In addition, you twice (ModeratorService and ArticleService) did not use AutoMapper in a similar situation

File: InternetHospital.BusinessLogic/Services/DoctorService.cs (hunk 3 above)
Comment (SergiyKozachok) [reply InternetHospital.BusinessLogic/Services/DoctorService.cs]: I deleted the method and moved the code to this method. AutoMapper does not work with IQueryable

**Review by voznovych — APPROVED** [review]

**Review by edward0802 — APPROVED** [review]

**Review by eugent482 — APPROVED** [review]

File: InternetHospital.WebApi/Controllers/DoctorsController.cs [hunk 4]
```diff
@@ -164,6 +164,95 @@ public IActionResult FillHistory([FromBody] IllnessHistoryModel illnessHistory)
...
+
+
+
+
+
+
+
+
+
+
```
Comment (voznovych) [inline InternetHospital.WebApi/Controllers/DoctorsController.cs]: Why so many empty lines?

File: InternetHospital.WebApi/Controllers/DoctorsController.cs [hunk 5]
```diff
@@ -145,6 +145,34 @@ public IActionResult FillHistory([FromBody] IllnessHistoryModel illnessHistory)
...
```
Comment (voznovych) [inline InternetHospital.WebApi/Controllers/DoctorsController.cs]: Why so many empty lines?

**Pull request merged**